websites/S/Sonic Retro/presence.ts

```diff
@@ -0,0 +1,279 @@
+				"Export pages - Sonic Retro",
+				"What links here - Sonic Retro",
+				"Create account - Sonic Retro",
+				"Log in - Sonic Retro",
+				"Reset password - Sonic Retro",
+				"Login required - Sonic Retro",
+				"Preferences - Sonic Retro",
+				"Watchlist - Sonic Retro",
+				"Reset tokens - Sonic Retro",
+				"Blocked users - Sonic Retro",
+				"File list - Sonic Retro"
+			];
+			presenceData.smallImageKey = "info_512";
+			presenceData.smallImageText = "info.sonicretro.org";
+			if (
+				pathname === "/" ||
+				pathname.includes("/Main_Page" || "/Main_Page#")
+			) {
+				presenceData.largeImageKey = "info";
+				presenceData.details = "Zooming through the main page";
+			} else if (pathname.includes("Special:")) {
+				if (unsupported.includes(document.title)) {
+					presenceData.largeImageKey = "logo_1024";
+					presenceData.details = "Viewing an unsupported page";
+				} else if (pathname === ("/Special:Search" || "/Special:Search#")) {
+					presenceData.largeImageKey = "search";
+					presenceData.details = "Searching for something...";
+				} else {
+					presenceData.largeImageKey = "category";
+					presenceData.details = "Browsing a special page";
+					presenceData.state = `${document.title.split(" - Sonic Retro")[0]}`;
+				}
+			} else if (document.title.startsWith("Category:")) {
+				presenceData.largeImageKey = "reading";
+				presenceData.details = "Reading an article:";
+				presenceData.state = `${
+					document.title.split("Category:")[1].split(" - Sonic Retro")[0]
+				}`;
+				presenceData.buttons = [
+					{
+						label: "View Article",
+						url: location.href
+					}
+				];
+			} else if (document.title.startsWith("Talk:")) {
+				presenceData.largeImageKey = "reading";
+				presenceData.details = "Browsing a talk page:";
+				presenceData.state = `${
+					document.title.split("Talk:")[1].split(" - Sonic Retro")[0]
+				}`;
+			} else if (document.title.startsWith("Sonic Retro:")) {
+				presenceData.largeImageKey = "reading";
+				presenceData.details = "Reading an article:";
+				presenceData.state = `${
+					document.title.split("Sonic Retro:")[1].split(" - Sonic Retro")[0]
+				}`;
+				presenceData.buttons = [
+					{
+						label: "View Article",
+						url: location.href
+					}
+				];
+			} else if (document.title.startsWith("User:")) {
+				presenceData.largeImageKey = "search";
+				presenceData.details = "Viewing a user's page:";
+				presenceData.state = `@${
+					document.title.split("User:")[1].split(" - Sonic Retro")[0]
+				}`;
+				presenceData.buttons = [
+					{
+						label: "View User",
+						url: location.href
+					}
+				];
+			} else if (
+				pathname.includes("/File:" || "/Sonic_Retro:General_disclaimer")
+			) {
+				presenceData.largeImageKey = "logo_1024";
+				presenceData.details = "Viewing an unsupported page";
+			} else if (
+				(pathname === "/index.php" && document.title.includes("Search")) ||
+				pathname === ("/Special:Search" || "/Special:Search#")
```

Comment:
this is invalid JS

Reply:
What I don't get is that not only does this work, prettier even asks to put it back once it's gone. 
![image](https://user-images.githubusercontent.com/96930330/169718305-ceee51cc-f0cd-4fe8-ae25-09f7422ba4d5.png)
So where do I go from here?